core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/cache/CacheServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2014-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.service.cache;
+
+import java.io.IOException;
+
+import org.hawkular.metrics.core.service.transformers.NumericDataPointCollector;
+import org.hawkular.metrics.datetime.DateTimeService;
+import org.hawkular.metrics.model.DataPoint;
+import org.hawkular.metrics.model.MetricId;
+import org.infinispan.Cache;
+import org.infinispan.commons.util.concurrent.NotifyingFuture;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.joda.time.Duration;
```

Comment:
I agree that we should switch to the new Time API in Java 8, but it is a low priority IMO. There are some critical functions we use for rounding dates/timestamps that we need to make sure are properly ported.